app.js

```diff
@@ -0,0 +1,102 @@
+// Update the following line with your API key:
+const API_KEY = 'YOUR_API_KEY_GOES_HERE';
+
+// -----------------------------------------------------------------------------
+// Delete the following lines of code when you've added your API key above:
+const error = 'Edit app.js to add your API key before running the app';
+alert(error);
+throw new Error(error);
+// -----------------------------------------------------------------------------
+
```

Comment:
Your 'add errors if API key is missing' commit removs the bit that actually initializes the client lib (`const ably = ...`).

Also, rather than requiring the customer to delete the lines once they add their api key, why not autodetect it -- e.g. `if (API_KEY.indexOf("YOUR") == 0) { ... }`? Then if that passes, initialize the client lib below (`const ably = new Ably.Realtime(API_KEY)`)

Reply:
Fixed.